xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/liveDataSource.js

```diff
@@ -51,7 +53,16 @@ define('xwiki-livedata-source', ['module', 'jquery'], function(module, $) {
     parameters.sort = liveDataQuery.sort.map(sort => sort.property);
     parameters.descending = liveDataQuery.sort.map(sort => sort.descending);
 
-    return Promise.resolve($.getJSON(entriesURL, $.param(parameters, true)).then(toLiveData));
+    if (entriesRequest) {
+      // Avoid ordering issues by cancelling outdated requests
+      // See https://jira.xwiki.org/browse/XWIKI-21447
```

Comment:
Probably better to explain with a few sentences the issue here, instead of pointing to a jira id.

Reply:
Yes, our comments are supposed to be self standing. An external link can be provided but if that links becomes broken in the future, the reader of the code need to understand what it was about and especially WHY that code was needed. Thx!